syft/messaging/message.py

```diff
@@ -189,72 +197,32 @@ def detail(worker: AbstractWorker, msg_tuple: tuple) -> "OperationMessage":
-        cmd_args = detailed_msg[2]
-        cmd_kwargs = detailed_msg[3]
+        detailed = sy.serde.msgpack.serde._detail(worker, action)
 
-        return OperationMessage(cmd_name, cmd_owner, cmd_args, cmd_kwargs, detailed_ids)
+        return OperationMessage(
+            detailed.name, detailed.operand, detailed.args, detailed.kwargs, detailed.return_ids
+        )
 
     @staticmethod
-    def bufferize(worker: AbstractWorker, operation: "OperationMessage") -> "OperationMessagePB":
+    def bufferize(
+        worker: AbstractWorker, operation_message: "OperationMessage"
+    ) -> "OperationMessagePB":
         """
         This function takes the attributes of a OperationMessage and saves them in Protobuf
         Args:
             worker (AbstractWorker): a reference to the worker doing the serialization
-            ptr (OperationMessage): an OperationMessage
+            operation_message (OperationMessage): an OperationMessage
         Returns:
             protobuf_obj: a Protobuf message holding the unique attributes of the message
         Examples:
             data = bufferize(message)
         """
         protobuf_op_msg = OperationMessagePB()
-        protobuf_op = OperationPB()
-        protobuf_op.command = operation.cmd_name
-
-        if type(operation.cmd_owner) == sy.generic.pointers.pointer_tensor.PointerTensor:
-            protobuf_owner = protobuf_op.owner_pointer
-        elif (
-            type(operation.cmd_owner)
-            == sy.frameworks.torch.tensors.interpreters.placeholder.PlaceHolder
-        ):
-            protobuf_owner = protobuf_op.owner_placeholder
-        else:
-            protobuf_owner = protobuf_op.owner_tensor
-
-        if operation.cmd_owner is not None:
-            protobuf_owner.CopyFrom(sy.serde.protobuf.serde._bufferize(worker, operation.cmd_owner))
-
-        if operation.cmd_args:
-            protobuf_op.args.extend(OperationMessage._bufferize_args(worker, operation.cmd_args))
-
-        if operation.cmd_kwargs:
-            for key, value in operation.cmd_kwargs.items():
-                protobuf_op.kwargs.get_or_create(key).CopyFrom(
-                    OperationMessage._bufferize_arg(worker, value)
-                )
-
-        if operation.return_ids is not None:
-            if type(operation.return_ids) == PlaceHolder:
-                return_ids = list((operation.return_ids,))
-            else:
-                return_ids = operation.return_ids
-
-            for return_id in return_ids:
-                if type(return_id) == PlaceHolder:
-                    protobuf_op.return_placeholders.extend(
-                        [sy.serde.protobuf.serde._bufferize(worker, return_id)]
-                    )
-                else:
-                    sy.serde.protobuf.proto.set_protobuf_id(protobuf_op.return_ids.add(), return_id)
-
-        protobuf_op_msg.operation.CopyFrom(protobuf_op)
+        protobuf_op = ComputationAction.bufferize(worker, operation_message.action)
```

Comment:
Is there an intentional design choice in "bufferize" being used functionally instead of as a method on operation_message.action?

Reply:
(not a big deal)

Reply:
Yes, there is, but it originally comes from msgpack serialization. If I recall correctly, it’s because there’s no object to call `detail`/`unbufferize` on; the object is the output of those methods (rather than an input.) As a result, those methods have to be static, and making `simplify`/`bufferize` static also creates a parallel structure for serialization in both directions.